### PR TITLE
Add regression_4000 tests

### PR DESCRIPTION
**Pull request by oleksiimoisieiev**

### CODE REVIEW

**Comment by dsemenets** [comment]
nist headers ported already. I put it to the include directory

File: src/regression_4000.c [hunk 1]
```diff
@@ -0,0 +1,5809 @@
...
+{
+	TEEC_Session session = { };
+	uint32_t ret_orig = 0;
+	size_t n = 0;
+	ADBG_STRUCT_DECLARE("Test TEE Internal API hash operations");
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(&c,
+		xtest_teec_open_session(&session, &crypt_user_ta_uuid, NULL,
+					&ret_orig)))
+		return;
```
Comment (dsemenets) [inline src/regression_4000.c:1026]: ADBG_Assert is necessary here

File: src/regression_4000.c [hunk 2]
```diff
@@ -0,0 +1,5809 @@
...
+	uint8_t out[64] = { };
+	size_t out_size = 0;
+	uint32_t ret_orig = 0;
+	size_t n = 0;
+	ADBG_STRUCT_DECLARE("Test TEE Internal API MAC operations");
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(&c,
+		xtest_teec_open_session(&session, &crypt_user_ta_uuid, NULL,
+					&ret_orig)))
+		return;
```
Comment (dsemenets) [inline src/regression_4000.c:1260]: Same as for line 1026

File: src/regression_4000.c [hunk 3]
```diff
@@ -0,0 +1,5809 @@
...
+
+	(void)ADBG_EXPECT_COMPARE_SIGNED(&c,
+		0, !=, memcmp(buf2, zeros, sizeof(buf2)));
+
+	(void)ADBG_EXPECT_COMPARE_SIGNED(&c,
+		0, !=, memcmp(buf2, buf1, sizeof(buf1)));
+out:
+	TEEC_CloseSession(&session);
+	ADBG_Assert(&c);
+	EndSubCase("TEE get random");
```
Comment (dsemenets) [inline src/regression_4000.c:2394]: Perhaps those lines should be in opposite order

File: src/regression_4000.c [hunk 4]
```diff
@@ -0,0 +1,5809 @@
...
+	size_t n = 0;
+	size_t m = 0;
+	size_t i = 0;
+
+	ADBG_STRUCT_DECLARE("Test TEE Internal API Bleichenbacher attack (negative)");
+
+	/* Setup session, initialize message to sign, create a keypair */
+	if (!ADBG_EXPECT_TEEC_SUCCESS(&c, xtest_teec_open_session(&s,
+			&crypt_user_ta_uuid, NULL, &ret_orig)))
+		return;
```
Comment (dsemenets) [inline src/regression_4000.c:5069]: ADBG_Assert missed

File: src/regression_4000.c (hunk 1 above)
Comment (oleksiimoisieiev) [reply src/regression_4000.c:1026]: +

File: src/regression_4000.c (hunk 2 above)
Comment (oleksiimoisieiev) [reply src/regression_4000.c:1260]: +

File: src/regression_4000.c (hunk 4 above)
Comment (oleksiimoisieiev) [reply src/regression_4000.c:5069]: +

**Pull request merged**